### ISSUE TITLE
ruby 2.3 url broken

### ISSUE THREAD

**Issue by mcandre** [rvm/rvm#3641]
Trace:

```
$ rvm get stable
Downloading https://get.rvm.io
No GPG software exists to validate rvm-installer, skipping.
Downloading https://github.com/rvm/rvm/archive/1.26.11.tar.gz
Downloading https://github.com/rvm/rvm/releases/download/1.26.11/1.26.11.tar.gz.asc
Found PGP signature at: 'https://github.com/rvm/rvm/releases/download/1.26.11/1.26.11.tar.gz.asc',
but no GPG software exists to validate it, skipping.

Upgrading the RVM installation in /Users/andrew.pennebaker/.rvm/
    RVM PATH line found in /Users/andrew.pennebaker/.mkshrc.
    RVM PATH line not found for Bash or Zsh, rerun this command with '--auto-dotfiles' flag to fix it.
    RVM sourcing line found in /Users/andrew.pennebaker/.bash_profile /Users/andrew.pennebaker/.zlogin.
Upgrade of RVM in /Users/andrew.pennebaker/.rvm/ is complete.

# Andrew Pennebaker,
#
#   Thank you for using RVM!
#   We sincerely hope that RVM helps to make your life easier and more enjoyable!!!
#
# ~Wayne, Michal & team.

In case of problems: http://rvm.io/help and https://twitter.com/rvm_io

Upgrade Notes:

/Users/andrew.pennebaker/.bashrc:1:PATH='/usr/bin:/bin:/usr/sbin:/sbin'

  * WARNING: Above files contains `PATH=` with no `$PATH` inside, this can break RVM,
    for details check https://github.com/wayneeseguin/rvm/issues/1351#issuecomment-10939525
    to avoid this warning append #PATH.


  * WARNING: you have GEM_HOME="/Users/andrew.pennebaker/.gem/ruby/2.3.0" this is conflicting with RVM, make sure to:

      unset GEM_HOME

  * No new notes to display.

RVM reloaded!
mcandre (master)$ rvm install ruby-2.3
Unknown ruby string (do not know how to handle): ruby-2.3.
Unknown ruby string (do not know how to handle): ruby-2.3.
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.11/x86_64/ruby-2.3.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Certificates in '/usr/local/etc/openssl/cert.pem' are already up to date.
Requirements installation successful.
Unknown ruby string (do not know how to handle): ruby-2.3.
Installing Ruby from source to: /Users/andrew.pennebaker/.rvm/rubies/ruby-2.3, this may take a while depending on your cpu(s)...
Unknown ruby string (do not know how to handle): ruby-2.3.
ruby-2.3 - #downloading ruby-2.3, this may take a while depending on your connection...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (22) The requested URL returned error: 404 Not Found
The requested url does not exist(22): 'http://cache.ruby-lang.org/pub/ruby/2.3/ruby-2.3.tar.bz2'
Checking fallback: http://ftp.ruby-lang.org/pub/ruby/2.3/ruby-2.3.tar.bz2
Checking fallback: http://www.mirrorservice.org/sites/ftp.ruby-lang.org/pub/ruby/2.3/ruby-2.3.tar.bz2
No fallback URL could be found, try increasing timeout with:

    echo "export rvm_max_time_flag=20" >> ~/.rvmrc

There has been an error fetching the ruby interpreter. Halting the installation.
```

**Comment by rys** [comment]
Can you update your RVM to current master and try again? From a quick look at the selector code, the bug shouldn't happen on the current master.

**Comment by mcandre** [comment]
Confirmed fixed in head.

bump that stable bb!

**Comment by fcalderan** [comment]
same error here: I've updated rvm today

```
> rvm get head

Downloading https://get.rvm.io
Downloading https://raw.githubusercontent.com/rvm/rvm/master/binscripts/rvm-installer.asc
Verifying /home/hart/.rvm/archives/rvm-installer.asc
gpg: Signature made Sat 31 Mar 2018 11:47:44 PM CEST using RSA key ID BF04FF17
gpg: Good signature from "Michal Papis (RVM signing) <mpapis@gmail.com>" [unknown]
gpg:                 aka "Michal Papis <michal.papis@toptal.com>" [unknown]
gpg:                 aka "[jpeg image of size 5015]" [unknown]
gpg: WARNING: This key is not certified with a trusted signature!
gpg:          There is no indication that the signature belongs to the owner.
Primary key fingerprint: 409B 6B17 96C2 7546 2A17  0311 3804 BB82 D39D C0E3
     Subkey fingerprint: 62C9 E5F4 DA30 0D94 AC36  166B E206 C29F BF04 FF17
GPG verified '/home/hart/.rvm/archives/rvm-installer'
Downloading https://github.com/rvm/rvm/archive/master.tar.gz
Upgrading the RVM installation in /home/hart/.rvm/
    RVM PATH line found in /home/hart/.mkshrc /home/hart/.profile /home/hart/.bashrc /home/hart/.zshrc.
    RVM sourcing line found in /home/hart/.profile /home/hart/.bash_profile /home/hart/.zlogin.
Upgrade of RVM in /home/hart/.rvm/ is complete.
  * No new notes to display.

RVM reloaded!```

and then I've tried to install ruby 2.6.0

```
> rvm install 2.6.0

Searching for binary rubies, this might take some time.
No binary rubies available for: debian/8/x86_64/ruby-2.6.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for debian.
Requirements installation successful.
Installing Ruby from source to: /home/hart/.rvm/rubies/ruby-2.6.0, this may take a while depending on your cpu(s)...
ruby-2.6.0 - #downloading ruby-2.6.0, this may take a while depending on your connection...
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (22) The requested URL returned error: 404 Not Found
The requested url does not exist(22): 'https://cache.ruby-lang.org/pub/ruby/2.6/ruby-2.6.0.tar.bz2'
Checking fallback: https://ftp.ruby-lang.org/pub/ruby/2.6/ruby-2.6.0.tar.bz2
No fallback URL could be found, try increasing timeout with:

    echo "export rvm_max_time_flag=20" >> ~/.rvmrc

There has been an error fetching the ruby interpreter. Halting the installation.```

Thanks.

**Comment by SarpongAbasimi** [comment]
@rys and @pkuczynski  I am having an issue with rvm .

chriss-MBP% rvm install 2.6.0  
Warning, new version of rvm available '1.29.8', you are using older version '1.29.7-next'.
You can disable this warning with:    echo rvm_autoupdate_flag=0 >> ~/.rvmrc
You can enable  auto-update  with:    echo rvm_autoupdate_flag=2 >> ~/.rvmrc
Searching for binary rubies, this might take some time.
No binary rubies available for: osx/10.13/x86_64/ruby-2.6.0.
Continuing with compilation. Please read 'rvm help mount' to get more information on binary rubies.
Checking requirements for osx.
Certificates bundle '/usr/local/etc/openssl@1.1/cert.pem' is already up to date.
Requirements installation successful.
Installing Ruby from source to: /Users/chrissongz/.rvm/rubies/ruby-2.6.0, this may take a while depending on your cpu(s)...
ruby-2.6.0 - #downloading ruby-2.6.0, this may take a while depending on your connection...
/Users/chrissongz/.rvm/scripts/functions/support: line 170: cd: /Users/chrissongz/.rvm/archives: No such file or directory
curl: (3) <url> malformed
There was an error(3).

No fallback URL can be calculated for /2.6/ruby-2.6.0.tar.bz2, open a ticket with full output here:

    https://github.com/rvm/rvm/issues


There has been an error fetching the ruby interpreter. Halting the installation.
chriss-MBP% rvm get head
Archives path '/Users/chrissongz/.rvm/archives' not writable, aborting.
chriss-MBP% rvm info

ruby-2.4.1:

  system:
    uname:        "Darwin chriss-MBP 17.7.0 Darwin Kernel Version 17.7.0: Thu Jun 21 22:53:14 PDT 2018; root:xnu-4570.71.2~1/RELEASE_X86_64 x86_64"
    name:         "OSX"
    version:      "10.13"
    architecture: "x86_64"
    bash:         "/bin/bash => GNU bash, version 3.2.57(1)-release (x86_64-apple-darwin17)"
    zsh:          "/bin/zsh => zsh 5.3 (x86_64-apple-darwin17.0)"
    remote_path:  "osx/10.13/x86_64"
    xcode:        "10.1"
cat: /Users/chrissongz/.rvm/installed.at: No such file or directory

  rvm:
    version:      "1.29.7-next ()"
    updated:      "can not read installation time"
    path:         "/Users/chrissongz/.rvm"
    autolibs:     "[4] Allow RVM to use package manager if found, install missing dependencies, install package manager (only OS X)."

  ruby:
    interpreter:  "ruby"
    version:      "2.4.1p111"
    date:         "2017-03-22"
    platform:     "x86_64-darwin15"
    patchlevel:   "2017-03-22 revision 58053"
    full_version: "ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin15]"

  homes:
    gem:          "/Users/chrissongz/.rvm/gems/ruby-2.4.1"
    ruby:         "/Users/chrissongz/.rvm/rubies/ruby-2.4.1"

  binaries:
    ruby:         "/Users/chrissongz/.rvm/rubies/ruby-2.4.1/bin/ruby"
    irb:          "/Users/chrissongz/.rvm/rubies/ruby-2.4.1/bin/irb"
    gem:          "/Users/chrissongz/.rvm/rubies/ruby-2.4.1/bin/gem"
    rake:         "/Users/chrissongz/.rvm/gems/ruby-2.4.1/bin/rake"

  environment:
    PATH:         "/Users/chrissongz/.rvm/gems/ruby-2.4.1/bin:/Users/chrissongz/.rvm/gems/ruby-2.4.1@global/bin:/Users/chrissongz/.rvm/rubies/ruby-2.4.1/bin:/Users/chrissongz/.rvm/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Library/Frameworks/Mono.framework/Versions/Current/Commands:/Applications/Xamarin Workbooks.app/Contents/SharedSupport/path-bin"
    GEM_HOME:     "/Users/chrissongz/.rvm/gems/ruby-2.4.1"
    GEM_PATH:     "/Users/chrissongz/.rvm/gems/ruby-2.4.1:/Users/chrissongz/.rvm/gems/ruby-2.4.1@global"
    MY_RUBY_HOME: "/Users/chrissongz/.rvm/rubies/ruby-2.4.1"
    IRBRC:        "/Users/chrissongz/.rvm/rubies/ruby-2.4.1/.irbrc"
    RUBYOPT:      ""
    gemset:       ""


chriss-MBP%
